server/tools/peertube-watch.ts

```diff
@@ -0,0 +1,56 @@
+import * as program from 'commander'
+import * as summon from 'summon-install'
+import * as path from 'path'
+import { execSync } from 'child_process'
+
+let videoURL
+
+program
+  .name('watch')
+  .arguments('<url>')
+  .option('-g, --gui <player>', 'player type', /^(airplay|stdout|chromecast|mpv|vlc|mplayer|ascii|xbmc)$/i, 'ascii')
+  .option('-i, --invert', 'invert colors (ascii player only)', true)
+  .on('--help', function () {
+    console.log()
+    console.log('  Available Players:')
+    console.log()
+    console.log('    - ascii')
+    console.log('    - mpv')
+    console.log('    - mplayer')
+    console.log('    - vlc')
+    console.log('    - stdout')
+    console.log('    - xbmc')
+    console.log('    - airplay')
+    console.log('    - chromecast')
+    console.log()
+    console.log('  Note: \'ascii\' is the only option not using WebTorrent and not seeding back the video.')
+    console.log()
+    console.log('  Examples:')
+    console.log()
+    console.log('    $ peertube watch -g mpv https://peertube.cpy.re/videos/watch/e8a1af4e-414a-4d58-bfe6-2146eed06d10')
+    console.log('    $ peertube watch --gui stdout https://peertube.cpy.re/videos/watch/e8a1af4e-414a-4d58-bfe6-2146eed06d10')
+    console.log('    $ peertube watch https://peertube.cpy.re/videos/watch/e8a1af4e-414a-4d58-bfe6-2146eed06d10')
+    console.log()
+  })
+  .action((url) => {
+    videoURL = url
+  })
+  .parse(process.argv)
+
+if (!videoURL) {
+  console.error('<url> field is required.')
+  process.exit(-1)
+} else { program['url'] = videoURL }
+
+handler(program)
+
+function handler (argv) {
+  if (argv['gui'] === 'ascii') {
+    const peerterminal = summon('peerterminal')
+    peerterminal([ '--link', videoURL, '--invert', argv['invert'] ])
+  } else {
+    summon('webtorrent-hybrid')
+    const CMD = 'node ' + path.resolve(__dirname, '../../../', 'node_modules', 'webtorrent-hybrid', 'bin', 'cmd.js')
+    execSync(CMD + ' --' + argv['gui'] + ' ' + argv['url'].replace('videos/watch', 'download/torrents') + '-720.torrent')
```

Comment:
Please create temp variable, this is difficult to read

Reply:
And why the `-720.torrent`?

Reply:
ah right, I forgot that hardcoded value